drivers/misc/optee_msg.h

```diff
@@ -0,0 +1,365 @@
+/****************************************************************************
+ * drivers/misc/optee_msg.h
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
```

Comment:
This is required to be documented in [LICENSE](https://github.com/apache/nuttx/blob/master/LICENSE).

Also, BSD-licensed code should be gated behind `ALLOW_BSD_COMPONENTS`